pkg/proxy/provider/http/http_test.go

```diff
@@ -0,0 +1,88 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+	"github.com/containous/traefik/v2/pkg/safe"
+	"github.com/containous/traefik/v2/pkg/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestProviderInit(t *testing.T) {
+	provider := &Provider{}
+	assert.Error(t, provider.Init())
+
+	provider = &Provider{
+		Endpoint: "localhost",
+	}
+	assert.NoError(t, provider.Init())
+}
+
+func TestGetDataFromEndpoint(t *testing.T) {
+	mux := http.NewServeMux()
+	server := httptest.NewServer(mux)
+
+	defer server.Close()
+	mux.HandleFunc("/endpoint", func(rw http.ResponseWriter, req *http.Request) {
+		_, _ = rw.Write([]byte("{OK}"))
+	})
+
+	provider := Provider{
+		Endpoint: server.URL + "/endpoint",
+	}
+
+	assert.NoError(t, provider.Init())
+
+	data, err := provider.getDataFromEndpoint(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, "{OK}", string(data))
+}
+
+func TestBuildConfiguration(t *testing.T) {
+	provider := Provider{
+		Endpoint: "http://127.0.0.1:9000/endpoint",
+	}
+
+	assert.NoError(t, provider.Init())
+
+	config := provider.buildConfiguration(context.Background(), []byte("{}"))
```

Comment:
Is this test really relevant? 😛 

Maybe we don't need to test this func since it only calls `json.Unmarshal` at the moment, wdyt?